### PR TITLE
fix/(TSEO-24): added thumbnailUrl in video Schema

### PR DESCRIPTION
**Pull request by virendrapatelnuk**
[Jira Link](https://nidigitalsolutions.jira.com/browse/TSEO-24?atlOrigin=eyJpIjoiNjE1Y2Q3MzQ0YjZhNGUyMThlODBkMGY5MWQxNGEyNjAiLCJwIjoiaiJ9)

### CODE REVIEW

File: packages/provider-queries/src/article_web.graphql [hunk 1]
```diff
@@ -99,6 +99,78 @@ fragment imageProps on Image {
     ratio
     url
   }
+  crop1024683: crop(ratio: "1024:683") {
```
Comment (flashcheeks) [inline packages/provider-queries/src/article_web.graphql:102]: @virendrapatelnuk can you explain why you are adding all of these crops to the article query?

Comment (virendrapatelnuk) [reply packages/provider-queries/src/article_web.graphql:102]: The latest articles have a common ratio 16:9 and 3:2 but for historic articles image ratios are different.
I analyzed all articles which had errors and found these common ratios.

PFA screenshots

image
image
image

Comment (virendrapatelnuk) [reply packages/provider-queries/src/article_web.graphql:102]: There were 123 articles which have errors in GSC.

Comment (virendrapatelnuk) [reply packages/provider-queries/src/article_web.graphql]: @flashcheeks I made changes please review.

Comment (virendrapatelnuk) [reply packages/provider-queries/src/article_web.graphql]: I analyzed, there are 80% of articles have a 3:2 ratio. so without changing the query we can solve the 80% of thumbnail issue.
Kindly review the changes. also, there is a separate [ticket](https://nidigitalsolutions.jira.com/secure/RapidBoard.jspa?rapidView=1409&modal=detail&selectedIssue=TSEO-28) for the fallback thumbnail.

**Comment by virendrapatelnuk** [comment]
@flashcheeks  I analyzed, there are 80% of articles have a 3:2 ratio. so without changing the query we can solve the 80% of thumbnail issue.
Kindly review the changes. also, there is a separate [ticket](https://nidigitalsolutions.jira.com/secure/RapidBoard.jspa?rapidView=1409&modal=detail&selectedIssue=TSEO-28) for the fallback thumbnail.

File: packages/article-skeleton/src/head.web.js [hunk 2]
```diff
@@ -126,19 +142,15 @@ function Head({ article, logoUrl, paidContentClassName }) {
...
+    getThumbnailUrl(article);
   const authors = getAuthorSchema(article);
   const caption = get(leadAsset, "caption", null);
   const title = headline || shortHeadline || "";
   const datePublished = new Date(publishedTime).toISOString();
   const dateModified = updatedTime || datePublished;
-  const thumbnailUrl = hasVideo
-    ? getVideoLeadAssetUrl(article)
-    : getThumbnailUrlFromImage(article);
+  const thumbnailUrl = getThumbnailUrl(article);
```
Comment (vinhlam) [inline packages/article-skeleton/src/head.web.js:153]: Are you sure removal of videoLeadAsset going to be ok? I've had times where I've had missing leadassets from videos unless you've added them to getThumbnail.

File: packages/article-skeleton/__tests__/web/head.web.test.js [hunk 3]
```diff
@@ -649,4 +649,50 @@ describe("Head", () => {
...
+  });
+
+  const ratios = [
+    { ratio: "1:1", crop: "crop11" },
+    { ratio: "2:3", crop: "crop23" },
+    { ratio: "3:2", crop: "crop32" },
+    { ratio: "16:9", crop: "crop169" },
+    { ratio: "4:5", crop: "crop45" },
+    { ratio: "1.25:1", crop: "crop1251" }
+  ];
```
Comment (vinhlam) [inline packages/article-skeleton/__tests__/web/head.web.test.js:671]: These do not seem to match your code below. `crop32 || crop1251 || crop11 || crop45 || crop23 || crop2251;`

Should `1.25:1` not be `5:4` ?

File: packages/article-skeleton/src/head.web.js (hunk 2 above)
Comment (virendrapatelnuk) [reply packages/article-skeleton/src/head.web.js:153]: I have not removed getVideoLeadAssetUrl(article), this is moved to getThumbnail method at line 109

File: packages/article-skeleton/__tests__/web/head.web.test.js (hunk 3 above)
Comment (virendrapatelnuk) [reply packages/article-skeleton/__tests__/web/head.web.test.js:671]: These crops are mentioned in graphql query in article_web.graphql, ratios array is raw data to create actual input for the test, input will look like following.

image

**Review by vinhlam — APPROVED** [review]

**Pull request merged**